docs/standard/serialization/binaryformatter-migration-guide/migrate-to-datacontractserializer.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "BinaryFormatter migration guide: Migrate to DataContractSerializer (XML)"
+description: "Migrate from BinaryFormatter to DataContractSerializer for XML serialization."
+ms.date: 5/31/2024
+no-loc: [BinaryFormatter, Serialization]
+dev_langs:
+  - CSharp
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "serialization [WCF]"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# Migrate to DataContractSerializer (XML)
+
+The .NET base class libraries provide two XML serializers: [XmlSerializer](../introducing-xml-serialization.md) and [DataContractSerializer](../../../fundamentals/runtime-libraries/system-runtime-serialization-datacontractserializer.md). There are some subtle differences between these two, but for the purpose of the migration, this section focuses only on `DataContractSerializer`. Why? Because it **fully supports the serialization programming model that was used by `BinaryFormatter`**. All the types that are already marked as `[Serializable]` or implement `ISerializable` can be serialized with `DataContractSerializer`. Where is the catch? Known types must be specified up front (that's why it's secure). You need to know them and be able to get the `Type`, **even for private types**.
```

Comment:
based on an offline feedback from @GrabYourPitchforks 

```suggestion
The .NET base class libraries provide two XML serializers: [XmlSerializer](../introducing-xml-serialization.md) and [DataContractSerializer](../../../fundamentals/runtime-libraries/system-runtime-serialization-datacontractserializer.md). There are some subtle differences between these two, but for the purpose of the migration, this section focuses only on `DataContractSerializer`. Why? Because it **fully supports the serialization programming model that was used by `BinaryFormatter`**. All the types that are already marked as `[Serializable]` or implement `ISerializable` can be serialized with `DataContractSerializer`. Where is the catch? Known types must be specified up front. You need to know them and be able to get the `Type`, **even for private types**.
```